### PR TITLE
ZCS-11866 : added rabbitmqctl

### PR DESCRIPTION
**Pull request by umagmrit**
Related PR's:
https://github.com/Zimbra/packages/pull/169
https://github.com/Zimbra/zm-build/pull/228

### CODE REVIEW

**Review by dasiyogesh — APPROVED** [review]

**Review by preshitaagrawal — APPROVED** [review]

File: src/bin/zmrabbitmqctl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+DAEMON=/opt/zimbra/common/sbin/${NAME}
+CONTROL=/opt/zimbra/common/sbin/rabbitmqctl
+DESC=rabbitmq
+USER=zimbra
+PID_FILE=/opt/zimbra/data/rabbitmq/pid
+LOCK_FILE=/opt/zimbra/data/rabbitmq/$NAME
+RABBITMQ_SERVER_CONSOLE_OUTPUT_DIR=/opt/zimbra/log/rabbitmq
+RABBITMQ_LOG_BASE=/opt/zimbra/log/rabbitmq
+RABBITMQ_STARTUP_TIMEOUT=600
+CONF_ENV_FILE=/opt/zimbra/common/lib/rabbitmq/rabbitmq-env.conf
```
Comment (silentsakky) [inline src/bin/zmrabbitmqctl:28]: aren't we storing conf files in /opt/zimbra/conf?

File: src/bin/zmrabbitmqctl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+# ***** END LICENSE BLOCK *****
+# 
+NAME=rabbitmq-server
+DAEMON=/opt/zimbra/common/sbin/${NAME}
+CONTROL=/opt/zimbra/common/sbin/rabbitmqctl
+DESC=rabbitmq
+USER=zimbra
+PID_FILE=/opt/zimbra/data/rabbitmq/pid
+LOCK_FILE=/opt/zimbra/data/rabbitmq/$NAME
+RABBITMQ_SERVER_CONSOLE_OUTPUT_DIR=/opt/zimbra/log/rabbitmq
```
Comment (silentsakky) [inline src/bin/zmrabbitmqctl:25]: not sure if we need folder for rabbitmq
we can use /opt/zimbra/log as path and rename files to rabbitmq.log and rabbitmq_err.log

File: src/bin/zmrabbitmqctl [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+# You should have received a copy of the GNU General Public License along with this program.
+# If not, see <https://www.gnu.org/licenses/>.
+# ***** END LICENSE BLOCK *****
+# 
+NAME=rabbitmq-server
+DAEMON=/opt/zimbra/common/sbin/${NAME}
+CONTROL=/opt/zimbra/common/sbin/rabbitmqctl
+DESC=rabbitmq
+USER=zimbra
+PID_FILE=/opt/zimbra/data/rabbitmq/pid
```
Comment (silentsakky) [inline src/bin/zmrabbitmqctl:23]: I thought pid files are stored in /opt/zimbra/log directory

File: src/bin/zmrabbitmqctl (hunk 2 above)
Comment (silentsakky) [reply src/bin/zmrabbitmqctl:25]: hmm there is another log file also https://github.com/Zimbra/packages/pull/169/files#diff-efca594a6905f9186d565e38fc00aa5078c5fa6add7085d27952b7e9cfc33490R6
a rethought, we can have one log file rabbitmq.log which will be saving from https://github.com/Zimbra/packages/pull/169/files#diff-efca594a6905f9186d565e38fc00aa5078c5fa6add7085d27952b7e9cfc33490R6
and another rabbitmq_startup.log and we can pipe stdout and stderr both outputs to this file, as logs coming from that command will be very less

File: src/bin/zmrabbitmqctl (hunk 3 above)
Comment (umagmrit) [reply src/bin/zmrabbitmqctl]: yes correct

File: src/bin/zmrabbitmqctl (hunk 1 above)
Comment (umagmrit) [reply src/bin/zmrabbitmqctl]: yes need store in /opt/zimbra/conf, updated changes for the same

File: src/bin/zmrabbitmqctl (hunk 2 above)
Comment (umagmrit) [reply src/bin/zmrabbitmqctl]: committed changes for same

File: src/bin/zmrabbitmqctl (hunk 1 above)
Comment (umagmrit) [reply src/bin/zmrabbitmqctl]: @silentsakky need one suggestion, currently we are getting /opt/zimbra/conf/rabbitmq-env.conf file from rabbitmq pacakge. In future if any changes introduced in rabbitmq-env.conf again we need to compile rabbitmq . I think this is not good approach.  In which repo we need to put rabbitmq-env.conf ?

**Review by zimsuchitgupta — APPROVED** [review]

File: src/bin/zmrabbitmqctl (hunk 1 above)
Comment (silentsakky) [reply src/bin/zmrabbitmqctl]: I don't think we have any repo where we can put conf files like that, so for now, let's go with current approach in future if need arises we will think about solving this problem

**Review by silentsakky — APPROVED** [review]

**Pull request merged**